core/src/test/scala/unit/kafka/coordinator/group/CoordinatorLoaderImplTest.scala

```diff
@@ -511,4 +544,35 @@ class CoordinatorLoaderImplTest {
 
     new FetchDataInfo(new LogOffsetMetadata(startOffset), fileRecords)
   }
+
+  private def logReadResult(
+    startOffset: Long,
+    producerId: Long,
+    producerEpoch: Short,
+    controlRecordType: ControlRecordType
+  ): FetchDataInfo = {
+    val fileRecords = mock(classOf[FileRecords])
+    val memoryRecords = MemoryRecords.withEndTransactionMarker(
+      startOffset,
```

Comment:
The marker created in https://github.com/apache/kafka/pull/14985/files#diff-252d8f54c521cc32e09406f127d568391346d3f92183fdb70f04108f4360f27fR197-R201 is then passed to the Partition and the Log. When the control record is finally written, it will get an offset, etc.